### PR TITLE
Button keyboard focus only

### PR DESCRIPTION
**Pull request by nathanbrud**
## What this PR does
Changes the default focus UI indication to be on keyboard events only.
Added unit tests for button component.

## Screenshots & Recordings 
https://github.com/vimeo/iris/assets/9106375/3dbcf4a7-3104-414b-b3f4-eea6a64f36ac

## How it does that 
The focus UI itself is being done using a component called `Focus`.
This component has a prop called `isKeyboardOnly` which supports the desired behavior.
Add a prop to button so it can be controlled and set the default to true.

## Testing 
* Click on a button using mouse, validate no focus border.
* Navigate to button using keyboard, validate focus is visible.

image

## Scanner Report
```
{
  "Button": {
    "instances": 785,
    "props": {
      "onClick": 645, // Tested
      "format": 479, // Tested
      "icon": 283, // Tested
      "size": 258, // Tested
      "variant": 239, // Tested
      "disabled": 175, // Tested
      "fluid": 112, // Cannot be tested no actual width for container and button so width comes as '' instead of 100%
      "data-testid": 69, // not relevant
      "loading": 61, // Tested
      "element": 55, // Tested
      "id": 54, // not relevant
      "iconPosition": 37, // Tested
      "style": 36, // not relevant
      "href": 29, // not relevant
      "status": 26, // Tested
      "ref": 22, // not relevant
      "aria-label": 21, // not relevant
      "className": 19, // not relevant
      "type": 15, // not relevant
      "key": 14, // not relevant
      "theme": 12, // not relevant
      "color": 9, // Tested
      "css": 8, // not relevant
      "pill": 8, // Tested
      "target": 7, // not relevant
      "tabIndex": 4, // not relevant
      "onMouseEnter": 3, // not relevant
      "onMouseLeave": 3, // not relevant
      "children": 2, // Tested
      "data-screen-recorder-cta": 2, // not relevant
      "alt": 1, // not relevant
      "data-js-modalCloseButton": 1, // not relevant
      "floating": 1, // not relevant, used only in 1 place
      "for": 1, // not relevant
      "title": 1 // not relevant
    }
  }
}
```

### CODE REVIEW

**Review by juliewongbandue — APPROVED** [review]
🎉

File: src/components/Button/Button.tsx [hunk 1]
```diff
@@ -111,7 +112,11 @@ function ButtonComponent({
         />
       )}
 
-      <Focus parent={ButtonStyled} radius={radius} />
+      <Focus
+        parent={ButtonStyled}
+        radius={radius}
+        isKeyboardOnly={isKeyboardOnlyFocus}
```
Comment (tylerthegrove) [inline src/components/Button/Button.tsx:118]: I don't know that we need a prop for this. I think we should just set this to true and remove the prop.

Comment (juliewongbandue) [reply src/components/Button/Button.tsx:118]: Ah okay. That makes sense to me. 👍

Comment (nathanbrud) [reply src/components/Button/Button.tsx:118]: Sure, removing

**Review by juliewongbandue — CHANGES_REQUESTED** [review]
A small change, as per Tyler's suggestion:  https://github.com/vimeo/iris/pull/269#discussion_r1200615067

**Review by juliewongbandue — APPROVED** [review]

**Pull request merged**